File: packages/galeria/ios/GaleriaModule.swift
import ExpoModulesCore

public class GaleriaModule: Module {
  public func definition() -> ModuleDefinition {
    Name("Galeria")

    View(GaleriaView.self) {
      Events("onIndexChange")

      OnViewDidUpdateProps { (view) in
        view.setupImageView()
      }

      Prop("urls") { (view, urls: [String]?) in
        view.urls = urls
      }
      
      Prop("items") { (view, items: [[String: Any]]?) in
        view.items = items
      }

      Prop("index") { (view, index: Int?) in
        view.initialIndex = index
      }

      Prop("galleryId") { (view, galleryId: String?) in
        view.galleryId = galleryId
      }

      Prop("theme") { (view, theme: Theme?) in
        view.theme = theme ?? .dark
      }
      Prop("closeIconName") { (view, closeIconName: String?) in
        view.closeIconName = closeIconName
      }
      Prop("rightNavItemIconName") { (view, rightNavItemIconName: String) in
        view.rightNavItemIconName = rightNavItemIconName
      }

    }
  }

  func onIndexChange(index: Int) {
    sendEvent("onIndexChange", ["currentIndex": index])
  }
}

File: packages/galeria/android/src/main/java/nandorojo/modules/galeria/GaleriaModule.kt
package nandorojo.modules.galeria

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class GaleriaModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("Galeria")

        View(GaleriaView::class) {
            Events(
                "onIndexChange"
            )
            Prop("theme") { view: GaleriaView, theme: Theme ->
                view.theme = theme
            }
            Prop("items") { view: GaleriaView, items: Array<Map<String, Any?>> ->
                view.items = items.toList()
            }
            Prop("urls") { view: GaleriaView, urls: Array<String> ->
                view.urls = urls
            }
            Prop("index") { view: GaleriaView, index: Int ->
                view.initialIndex = index
            }
            Prop("galleryId") { view: GaleriaView, galleryId: String? ->
                view.galleryId = galleryId
            }
            Prop("disableHiddenOriginalImage") { view: GaleriaView, disableHiddenOriginalImage: Boolean ->
                view.disableHiddenOriginalImage = disableHiddenOriginalImage
            }
            Prop("edgeToEdge") { view: GaleriaView, edgeToEdge: Boolean ->
                view.edgeToEdge = edgeToEdge
            }
            Prop("transitionOffsetY") { view: GaleriaView, transitionOffsetY: Int? ->
                view.transitionOffsetY = transitionOffsetY
            }
            Prop("transitionOffsetX") { view: GaleriaView, transitionOffsetX: Int? ->
                view.transitionOffsetX = transitionOffsetX
            }
        }
    }
}
